### ISSUE TITLE
Cypress show runtime error when use next.js with @cypress/react/plugins/next plugin

### ISSUE THREAD

**Issue by jyunhanlin** [cypress-io/cypress#15734]
### Current behavior

Use cypress with the `@cypress/react/plugins/next` plugin.
When run cypress, it shows the errors
```
TypeError: Cannot read property 'length' of undefined
    at getBaseWebpackConfig (/Users/Hank.Lin/playground/nextjs-starter-app/node_modules/next/dist/build/webpack-config.js:8:575)
    at getNextWebpackConfig (/Users/Hank.Lin/playground/nextjs-starter-app/node_modules/@cypress/react/plugins/next/file-preprocessor.js:16:35)
    at async getCachedWebpackPreprocessor (/Users/Hank.Lin/playground/nextjs-starter-app/node_modules/@cypress/react/plugins/next/file-preprocessor.js:59:26)
    at async Object.handler (/Users/Hank.Lin/playground/nextjs-starter-app/node_modules/@cypress/react/plugins/next/file-preprocessor.js:70:26)
```

### Desired behavior

It should run the tests correctly.

### Test code to reproduce

install steps
```sh
npx create-next-app nextjs-starter-app
npm install --save-dev cypress @cypress/react
```

in cypress/plugins/index.js
```js
module.exports = (on, config) => {
  require('@cypress/react/plugins/next')(on, config);
}
```

### Versions
cypress: 6.8.0
@cypress/react: 4.6.4
next: 10.1.2

**Comment by jyunhanlin** [comment]
I found a tricky method to make cypress run correctly.

steps: 
1. install webpack version 4
```sh
npm install -D webpack@4
```

2. go to /node_modules/@cypress/react/plugins/next/file-preprocessor.js, and change the `rewrites` manually as below
```js
  const nextWebpackConfig = await getNextJsBaseWebpackConfig(
    config.projectRoot,
    {
      buildId: `@cypress/react-${Math.random().toString()}`,
      config: nextConfig,
      dev: false,
      isServer: false,
      pagesDir: config.projectRoot,
      entrypoints: {},
      rewrites: {
        fallback:[],
        afterFiles: [],
        beforeFiles: [],
      },
    },
  )
```

**Comment by dmtrKovalenko** [comment]
This already has a fix here #15708 and should be released as a part of 7.0 today

**Comment by JessicaSachs** [comment]
Closing this issue now. Please reopen with any failures.
